mvc/src/main/java/cz/muni/fi/controllers/DriverController.java

```diff
@@ -0,0 +1,131 @@
+        log.info("Create new driver.");
+        DriverDTO driver = new DriverDTO();
+        model.addAttribute("driver", driver);
+        return "driver/edit";
+    }
+    
+    /**
+     * Prepares form for editing.
+     * 
+     * @param id of driver to edit
+     * @param model data to display
+     * @return JSP page
+     */
+    @RequestMapping(value = "/edit/{id}", method = RequestMethod.GET)
+    public String editDriver(@PathVariable Long id, Model model) {
+        log.info("Edit driver.");
+        model.addAttribute("driver", driverFacade.getDriverByID(id));
+        return "driver/edit";
+    }
+    
+    /**
+     * Create/update driver to database.
+     * 
+     * @param driverDTO to be created/edited
+     * @param model data to display
+     * @param redirectAttributes to show result of method
+     * @param uriBuilder uri
+     * @return JSP page
+     */
+    @RequestMapping(value = "/save", method = RequestMethod.POST)
+    public String submitEdit(@Valid @ModelAttribute("driver") DriverDTO driverDTO,
+                             Model model,
+                             RedirectAttributes redirectAttributes,
+                             UriComponentsBuilder uriBuilder) {
+
+        log.info("Saving DriverDTO: {}", driverDTO);
+        if(driverDTO.getName().equals("") || driverDTO.getSurname().equals("") || driverDTO.getNationality().equals("")){
+            StringBuilder builder = new StringBuilder("Please fill all values!");
+
+            model.addAttribute("alert_danger", builder.toString());
+            return "driver/edit";
+        }
+        if(driverDTO.getId() == null){
+            DriverCreateDTO driver = new DriverCreateDTO();
+            driver.setName(driverDTO.getName());
+            driver.setSurname(driverDTO.getSurname());
+            driver.setNationality(driverDTO.getNationality());
+            driver.setSpecialSkill(driverDTO.getSpecialSkill());
+            driverFacade.createDriver(driver);
+            redirectAttributes.addFlashAttribute("alert_success", "Driver was created.");
+        } else{
+            driverFacade.updateDriver(driverDTO);
+            redirectAttributes.addFlashAttribute("alert_success", "Driver details were saved.");
+        }
+        return "redirect:" + uriBuilder.path("/driver/").build().toUriString();
+    }
+
+    /**
+     * Delete driver.
+     * 
+     * @param id to delete
+     * @param model data to display
+     * @param redirectAttributes to show result of method
+     * @param uriBuilder uri
+     * @return JSP page
+     */
+    @RequestMapping(value = "/delete/{id}", method = RequestMethod.POST)
+    public String delete(@PathVariable long id,
+                         Model model,
+                         RedirectAttributes redirectAttributes,
+                         UriComponentsBuilder uriBuilder) {
+        log.info("Deleting driver with id: {}", id);
+        DriverDTO driverDTO = driverFacade.getDriverByID(id);
+        try {
```

Comment:
What is a usecase of this situation in try catch?

Reply:
If driver is main driver in some car, DAO throws SQLException and in that case driver cannot be deleted, because car wouldn't have driver.

Reply:
okay, thanks for explanation.